### ISSUE TITLE
Formatter removes EEx comment tags

### ISSUE THREAD

**Issue by mulias** [feliperenan/heex_formatter#45]
After running the formatter on this code
```
    <div>
      <%# some %>
      <%# comment %>
      <%# lines %>
    </div>
```

The result has all comment tags stripped out
```
    <div></div>
```

Putting a comment inside of an inline expression tag also doesn't quite work, the content and all following tags get un-indented one level
```
    <div>
    <% # some %>
    <% # comment %>
    <% # lines %>
  </div>
  <.next_thing>
```

A workaround for now is to use a replace expression tag. This stays the same after running the formatter
```
    <div>
      <%= # some %>
      <%= # comment %>
      <%= # lines %>
    </div>
```

**Comment by feliperenan** [comment]
Hi @mulias 

Thanks for reporting this. I'm working in a refactor to use Inspect.Algebra module to format the code. Hopefully this issue will be solved once I'm done with that.

**Comment by 03juan** [comment]
Unfortunately it's stripping out the comments from the original code and now is also inserting an indented blank line

```html
<div>
  
</div>
```

**Comment by josevalim** [comment]
This will have to be fixed in Elixir and therefore wait for a new Elixir version. :(

@feliperenan let me know if you want to contribute this feature upstream!

**Comment by feliperenan** [comment]
I do!😊. I will probably start to take a look at from tomorrow's on.

**Comment by mulias** [comment]
Quick update, since my initial description is no longer correct

case 1 input:
```
    <div>
      <%# some %>
      <%# comment %>
      <%# lines %>
    </div>
```
output:
```
  <div>
    
  </div>
```

case 2 input:
```
    <div>
      <% # some %>
      <% # comment %>
      <% # lines %>
    </div>
```
output:
```
  <div>
    <%  %>
    <%  %>
    <%  %>
  </div>
```

case 3 input:
```
    <div>
      <%= # some %>
      <%= # comment %>
      <%= # lines %>
    </div>
```
output:
```
  <div>
    <%=  %>
    <%=  %>
    <%=  %>
  </div>
```

The workaround mentioned above no longer works, but everything else seems to be working great!

**Comment by josevalim** [comment]
@feliperenan we need to call `Code.string_to_quoted_with_comments` and then pass the `:comments` option in `Code.quoted_to_algebra`. (This is separate from the original issue).

**Comment by feliperenan** [comment]
The main issue is now fixed on main but depends on Elixir 1.14 that is yet to be released.

PS: You should comment your code using `<%!-- comment --%>` syntax because `<%# comment %>` will be deprecated soon. The previous work around should work.
